src/ducks/transactions/actions/BillAction.jsx

```diff
@@ -45,27 +45,29 @@ const getBill = (transaction, actionProps) => {
 class AugmentedModalButton extends React.Component {
   state = { opened: false }
 
-  open() {
+  open = event => {
+    event.stopPropagation()
     this.setState({ opened: true })
   }
 
-  close() {
+  close = event => {
+    event.stopPropagation()
     this.setState({ opened: false })
   }
 
   render() {
     const { fileId, text, compact } = this.props
     return (
       <ButtonAction
-        onClick={() => this.open()}
+        onClick={this.open}
```

Comment:
Pourquoi ici on a pas besoin de mettre dans le constructeur: `this.open = this.open.bind(this)` ?

Reply:
Parce qu'on utilise une "class property" (aussi appelé "field declaration" sur MDN : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Field_declarations), avec une arrow function. Du coup le `this` de la méthode sera toujours "le bon" (l'objet qui porte la méthode). C'est pas encore dans le standard mais Babel le transpile : https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEBiD29oG8BQ1oCMwC9oF5oAKASgID5ph4A7CeEAUwDoR4BzIgFwAsBLCCVQBfIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Ces2016%2Ces2017%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=

AFAIK ça revient au même que de faire un `bind(this)` sur la méthode dans le constructeur.